doc/examples/type_hints.rst

```diff
@@ -98,27 +98,88 @@ These methods automatically add an "_id" field.
 
 .. doctest::
 
-  >>> from typing import TypedDict
+  >>> # The typing_extensions package must be installed unless using Python 3.11+.
+  >>> from typing_extensions import TypedDict
```

Comment:
We should use `:pyversion: >= 3.8` instead of `typing_extensions` in this example